File: src/main/java/ru/vmsystems/template/domain/shared/Role.java
package ru.vmsystems.template.domain.shared;

public enum Role {

    ROLE_SUPER_ADMIN("ROLE_SUPER_ADMIN", "Супер админ"),
    ROLE_ADMIN("ROLE_ADMIN", "Администратор"),
    ROLE_USER("ROLE_USER", "Пользователь");

    private String role;
    private String comment;

    Role(String role, String comment) {
        this.role = role;
        this.comment = comment;
    }

    @Override
    public String toString() {
        return role;
    }

    public String getComment() {
        return comment;
    }
}

File: src/main/java/ru/vmsystems/glasscloud/config/security/Role.kt
package ru.vmsystems.glasscloud.config.security

enum class Role(private val role: String, val comment: String) {

    ROLE_USER("ROLE_USER", "Пользователь"),
    ROLE_ADMIN("ROLE_ADMIN", "Администратор");

    companion object {
        fun name(role: Role): String {
            return role.role
        }
    }
}
